docs/source/serialization.mdx

```diff
@@ -49,255 +60,365 @@ Ready-made configurations include the following models:
+Hugging Face Hub or one that's stored locally.
+
+The resulting `model.onnx` file can then be run on one of the [many
+accelerators](https://onnx.ai/supported-tools.html#deployModel) that support the
+ONNX standard. For example, we can load and run the model with [ONNX
+Runtime](https://onnxruntime.ai/) as follows:
 
 ```python
-import onnxruntime as ort
+from transformers import AutoTokenizer
+from onnxruntime import InferenceSession
+
+tokenizer = AutoTokenizer.from_pretrained("distilbert-base-uncased")
+session = InferenceSession("onnx/model.onnx")
+# ONNX Runtime expects NumPy arrays as input
+inputs = tokenizer("Using DistilBERT with ONNX Runtime!", return_tensors="np")
+outputs = session.run(output_names=["last_hidden_state"], input_feed=dict(inputs))
+```
 
-from transformers import BertTokenizerFast
-tokenizer = BertTokenizerFast.from_pretrained("bert-base-cased")
+The required output names (i.e. `["last_hidden_state"]`) can be obtained by
+taking a look at the ONNX configuration of each model. For example, for
+DistilBERT we have:
 
-ort_session = ort.InferenceSession("onnx/bert-base-cased/model.onnx")
+```python
+from transformers.models.distilbert import DistilBertConfig, DistilBertOnnxConfig
 
-inputs = tokenizer("Using BERT in ONNX!", return_tensors="np")
-outputs = ort_session.run(["last_hidden_state", "pooler_output"], dict(inputs))
+config = DistilBertConfig()
+onnx_config = DistilBertOnnxConfig(config)
+print(list(onnx_config.outputs.keys()))
 ```
 
-The outputs used (`["last_hidden_state", "pooler_output"]`) can be obtained by taking a look at the ONNX
-configuration of each model. For example, for BERT:
+which outputs:
 
 ```python
-from transformers.models.bert import BertOnnxConfig, BertConfig
-
-config = BertConfig()
-onnx_config = BertOnnxConfig(config)
-output_keys = list(onnx_config.outputs.keys())
+['last_hidden_state']
 ```
 
-#### Implementing a custom configuration for an unsupported architecture
+### Selecting features for different model topologies
 
-Let's take a look at the changes necessary to add a custom configuration for an unsupported architecture. Firstly, we
-will need a custom ONNX configuration object that details the model inputs and outputs. The BERT ONNX configuration is
-visible below:
+Each ready-made configuration comes with a set of _features_ that enable you to
+export models for different types of topologies or tasks. As shown in the table
+below, each feature is associated with a different auto class:
 
-```python
-class BertOnnxConfig(OnnxConfig):
-    @property
-    def inputs(self) -> Mapping[str, Mapping[int, str]]:
-        return OrderedDict(
-            [
-                ("input_ids", {0: "batch", 1: "sequence"}),
-                ("attention_mask", {0: "batch", 1: "sequence"}),
-                ("token_type_ids", {0: "batch", 1: "sequence"}),
-            ]
-        )
+| Feature                              | Auto Class                           |
```

Comment:
We might want to automatically populate this table based on the state of `FeaturesManager`. I've left it as manual for now.